File: BSDN-API/src/main/java/com/gxu/bsdn/entity/Article.java
package com.gxu.bsdn.entity;


import java.io.Serializable;
import java.util.Date;
import lombok.Data;


@Data
public class Article implements Serializable {

    private Long id;


    private String title;


    private Date createTime;


    private Long authorId;


    private Long categoryId;


    private Long viewCount;


    private Long thumbCount;


    private Long collectCount;


    private String content;


    private String contentHtml;

    private static final long serialVersionUID = 1L;
}

File: BSDN-User/vite-vue-ts/src/Type/api/article.ts
import {User} from "@/Type/api/user";
import {Category} from "@/Type/api/category";

export interface ArticleWithOther {
  id: number,
  title: string,
  createTime: string,
  authorId: number,
  categoryId: number,
  viewCount: number,
  thumbCount: number,
  collectCount: number,
  content: string,
  contentHtml: string,
  author: User,
  category: Category
}

export interface PublicArticle {
  id: number,
  title: string,
  createTime: string,
  authorId: number,
  categoryId: number,
  viewCount: number,
  thumbCount: number,
  collectCount: number,
  content: string,
  contentHtml: string
}
export interface ArticleParam{
  author: string,
  title: string,
  category: string
}
